### PR TITLE
Add the matrix strategy to workflow to make integration tests faster

### PR DESCRIPTION
**Pull request by maxday**
### What does this PR do?

This PR should make the integration testing step faster by using the matrix strategy, it will run tests in parallel.

### Motivation

Increase productivity by reducing waiting time

### Testing Guidelines

Codebase was not change but a refactoring for integration testing has been made. This should not break the integration tests and should be fully backward compatible

### Additional Notes

Current average workflow time for the integration tests step : between 13 and 17 minutes
With this PR, the new time is about 8min, see here : https://github.com/DataDog/datadog-lambda-js/actions/runs/771988973

### Types of Changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: scripts/build_layers.sh [hunk 1]
```diff
@@ -10,7 +10,6 @@ set -e
 
 LAYER_DIR=".layers"
 LAYER_FILES_PREFIX="datadog_lambda_node"
-NODE_VERSIONS=("10.15" "12.13" "14.15")
```
Comment (DarcyRaynerDD) [inline scripts/build_layers.sh:13]: We run this locally while deploying, so I think we probably want to keep the ability to build all layers in a single command. We could have an env var or arg that limits the script to a single runtime. You can see an example of how we do that in the [./publish_layers.sh](https://github.com/DataDog/datadog-lambda-js/blob/main/scripts/publish_layers.sh#L28) script

Comment (maxday) [reply scripts/build_layers.sh:13]: thanks, this should be resolved with 944c85e

**Comment by nhinsch** [comment]
It looks like the integration tests (and other CI jobs in the same file) are not running on this PR -- not sure why that is but we should look into that...

File: scripts/run_integration_tests.sh [hunk 2]
```diff
@@ -65,49 +64,48 @@ function remove_stack() {
 }
 trap remove_stack EXIT
 
-echo "Deploying functions"
+echo "Deploying functions for runtime $SERVERLESS_RUNTIME"
```
Comment (nhinsch) [inline scripts/run_integration_tests.sh:67]: Nit: I would add a check with a helpful error message in case we forget to specify `$SERVERLESS_RUNTIME`

**Comment by maxday** [comment]
> It looks like the integration tests (and other CI jobs in the same file) are not running on this PR -- not sure why that is but we should look into that...

Yes, we may want to modify the "on:" trigger in the build.yml file to **push** + **pull_request** ?

**Comment by nhinsch** [comment]
> > It looks like the integration tests (and other CI jobs in the same file) are not running on this PR -- not sure why that is but we should look into that...
> 
> Yes, we may want to modify the "on:" trigger in the build.yml file to **push** + **pull_request** ?

I don't thinnnnk that's the issue. This has been working fine on previous PRs ([example](https://github.com/DataDog/datadog-lambda-js/pull/179)) as far as I can tell.

**Comment by nhinsch** [comment]
Let's say I want to run these tests locally. My understanding is that I now need to pass in three new environment variables:

`RUNTIME=xxx NODE_VERSION=xxx SERVERLESS_RUNTIME=xxx ...run_integration_tests.sh`

In addition to the existing `API_KEY` and `UPDATE_SNAPSHOTS` environment variables.

The formats for the three environment variables are different, but fundamentally all three hold the same value. For example, if I want to test Node 10.x, I need to pass `RUNTIME=node10 NODE_VERSION=10.5 SERVERLESS_RUNTIME=node10.x`.

I think it would be more user-friendly to just pass `NODE_VERSION=10` and then have the script use that single variable to determine the correct runtime, node_version, and serverless_runtime based on a hardcoded map. What do you think?

File: scripts/run_integration_tests.sh [hunk 3]
```diff
@@ -11,7 +11,6 @@ set -e
 # These values need to be in sync with serverless.yml, where there needs to be a function
```
Comment (nhinsch) [inline scripts/run_integration_tests.sh:11]: I would update the comment at the top of this file to document the new expected environment variable(s)

**Comment by maxday** [comment]
> Let's say I want to run these tests locally. My understanding is that I now need to pass in three new environment variables:
> 
> `RUNTIME=xxx NODE_VERSION=xxx SERVERLESS_RUNTIME=xxx ...run_integration_tests.sh`
> 
> In addition to the existing `API_KEY` and `UPDATE_SNAPSHOTS` environment variables.
> 
> The formats for the three environment variables are different, but fundamentally all three hold the same value. For example, if I want to test Node 10.x, I need to pass `RUNTIME=node10 NODE_VERSION=10.5 SERVERLESS_RUNTIME=node10.x`.
> 
> I think it would be more user-friendly to just pass `NODE_VERSION=10` and then have the script use that single variable to determine the correct runtime, node_version, and serverless_runtime based on a hardcoded map. What do you think?

Great idea ! I'll do it

**Comment by nhinsch** [comment]
Another thought: can we also keep the option to run all of the integration tests? In case I want to run them all locally and not test one runtime at a time.

**Comment by nhinsch** [comment]
> It looks like the integration tests (and other CI jobs in the same file) are not running on this PR -- not sure why that is but we should look into that...

Per our Slack convo this is because this PR is opened on a forked repo -- let's switch over to the main repo at some point so we can see how everything works in practice.

**Pull request closed without merging**